### DISCUSSION TITLE
Clashing usernames on different 3rd party providers

### DISCUSSION THREAD

**Discussion by andrewsrichardson** [strapi/strapi#8372] [Legacy Topics]
#4998 referenced this issue, but I believe nothing was changed.

In the current system, two different users who share the same email prefix (i.e. andrew@gmail.com and andrew@hotmail.com) will cause errors when trying to sign up as their username will clash.

I am trying to implement a workaround here in which a user must specify a username before they sign up, but the 3rd party auth callback does not seem to take a username field. 

Does anyone know what I should be editing in the user-permissions plugin in order to change how the username for a 3rd party auth user is created?

Or rather than change the plugins, perhaps any other workarounds.

Thanks

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question, we are currently planning to decomission our GitHub Discussions in early 2021 and we don't want to lose the great questions asked by our community.

Questions should be directed to [our forum](https://forum.strapi.io/). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).
